pkg/asset/ignition/bootstrap/registries.go

```diff
@@ -1,35 +1,42 @@
 package bootstrap
 
 import (
-	"k8s.io/apimachinery/pkg/util/sets"
-
+	configv1 "github.com/openshift/api/config/v1"
 	"github.com/openshift/installer/pkg/types"
 )
 
+// sourceSetKey represents the set of fields that have to be unique to form
+// a merged list without duplicate entries for Image sources.
+type sourceSetKey struct {
+	Source       string
+	SourcePolicy configv1.MirrorSourcePolicy
+}
+
 // MergedMirrorSets consolidates a list of ImageDigestSources so that each
 // source appears only once.
 func MergedMirrorSets(sources []types.ImageDigestSource) []types.ImageDigestSource {
-	sourceSet := make(map[string][]string)
-	mirrorSet := make(map[string]sets.String)
-	orderedSources := []string{}
+	sourceSet := make(map[sourceSetKey][]string)
+	mirrorSet := make(map[sourceSetKey][]string)
```

Comment:
I guess we still need to use `map[string]sets.String` in order to make sure all mirrors are unique like before?

Reply:
Using Set causes SA errors. My fix for linter errors broke this and I forgot to rerun the tests. Working on a fix.